scripts/cmake/z_vcpkg_fixup_rpath_macho.cmake

```
@@ -0,0 +1,154 @@
+function(z_vcpkg_calculate_corrected_macho_rpath)
+    cmake_parse_arguments(PARSE_ARGV 0 "arg"
+      ""
+      "MACHO_FILE_DIR;OUT_NEW_RPATH_VAR"
+      "")
+
+    set(current_prefix "${CURRENT_PACKAGES_DIR}")
+    set(current_installed_prefix "${CURRENT_INSTALLED_DIR}")
+    file(RELATIVE_PATH relative_from_packages "${CURRENT_PACKAGES_DIR}" "${arg_MACHO_FILE_DIR}")
+    if("${relative_from_packages}/" MATCHES "^debug/" OR "${relative_from_packages}/" MATCHES "^(manual-)?tools/.*/debug/.*")
+        set(current_prefix "${CURRENT_PACKAGES_DIR}/debug")
+        set(current_installed_prefix "${CURRENT_INSTALLED_DIR}/debug")
+    endif()
+
+    # compute path relative to lib
+    file(RELATIVE_PATH relative_to_lib "${arg_MACHO_FILE_DIR}" "${current_prefix}/lib")
+    # remove trailing slash
+    string(REGEX REPLACE "/+$" "" relative_to_lib "${relative_to_lib}")
+
+    if(NOT relative_to_lib STREQUAL "")
+        set(new_rpath "@loader_path/${relative_to_lib}")
+    else()
+        set(new_rpath "@loader_path")
+    endif()
+
+    set("${arg_OUT_NEW_RPATH_VAR}" "${new_rpath}" PARENT_SCOPE)
+endfunction()
+
+function(z_vcpkg_fixup_macho_rpath_in_dir)
+    # We need to iterate through everything because we
+    # can't predict where a Mach-O file will be located
+    file(GLOB root_entries LIST_DIRECTORIES TRUE "${CURRENT_PACKAGES_DIR}/*")
+
+    # Skip some folders for better throughput
+    list(APPEND folders_to_skip "include")
+    list(JOIN folders_to_skip "|" folders_to_skip_regex)
+    set(folders_to_skip_regex "^(${folders_to_skip_regex})$")
+
+    find_program(
+        install_name_tool_cmd
+        NAMES install_name_tool
+        DOC "Absolute path of install_name_tool cmd"
+        REQUIRED
+    )
+
+    find_program(
+        otool_cmd
+        NAMES otool
+        DOC "Absolute path of otool cmd"
+        REQUIRED
+    )
+
+    find_program(
+        file_cmd
+        NAMES file
+        DOC "Absolute path of file cmd"
+        REQUIRED
+      )
+
+    foreach(folder IN LISTS root_entries)
+        if(NOT IS_DIRECTORY "${folder}")
+            continue()
+        endif()
+
+        get_filename_component(folder_name "${folder}" NAME)
+        if(folder_name MATCHES "${folders_to_skip_regex}")
+            continue()
+        endif()
+
+        file(GLOB_RECURSE macho_files LIST_DIRECTORIES FALSE "${folder}/*")
+        list(FILTER macho_files EXCLUDE REGEX "\\\.(cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")

```

It seems like 3 `\`s can't be the intent here? It looks like the intent is to match extension which would look like this.

```suggestion
        list(FILTER macho_files EXCLUDE REGEX [[\.(cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$]])
```

---

Copied from the linux equivalent, probably also needs to be adjusted there https://github.com/microsoft/vcpkg/blob/a59ba45e60167addb90e2aa7b285fc16ec8ba38a/scripts/cmake/z_vcpkg_fixup_rpath.cmake#L88